libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -177,6 +183,11 @@ void ImmutableValidator::analyseVariableReference(VariableDeclaration const& _va
 	if (!_variableReference.isStateVariable() || !_variableReference.immutable())
 		return;
 
+	auto secondaryLocation = SecondarySourceLocation().append(
+		"While constructing contract '" + m_currentContract.name() + "'",
```

Comment:
Won't this also be added in deployed functions? In general, I don't find this particularly helpful. Which situations do you imagine this to be useful?

Reply:
We have one new error with identical messages but different contracts and Daniel suggested I'd add that to differentiate better:

```
contract B {
    uint immutable private x;

    constructor() {
    }

    function f() internal view virtual returns(uint) { return 1; }
    function readX() internal view returns(uint) { return x; }
}

contract C is B {
    uint immutable y;
    constructor() {
        y = 3;
    }
    function f() internal view override returns(uint) { return readX(); }

}
// ----
// TypeError 7733: (196-197): Immutable variables cannot be read before they are initialized.
// TypeError 2658: (0-202): Construction control flow ends without initializing all immutable state variables.
// TypeError 7733: (196-197): Immutable variables cannot be read before they are initialized.
// TypeError 2658: (204-361): Construction control flow ends without initializing all immutable state variables.
```

Reply:
I don't see how the errors here are correct - see https://github.com/ethereum/solidity/pull/11750#pullrequestreview-732864091

Reply:
Ok, maybe the secondary location makes sense, but only if the read is not in the current contract.